### ISSUE TITLE
port STIR TOF support to SIRF

### ISSUE THREAD

**Issue by KrisThielemans** [SyneRBI/SIRF#315]
we're finalising the STIR TOF PR https://github.com/UCL/STIR/pull/304

This should normally require very few changes to SIRF. I think the only places are where we expose getting the projection data as TOF adds another dimension.

I think this only affects `PETAcquisitionData`.`dimensions`, `as_array` and `show`. @evgueni-ovtchinnikov could you please confirm?

Ideally, we should prepare for this in 2.0 by already letting these functions return the TOF dimension (although it's of course size 1). That way, we can enable TOF support in 2.1, without breaking backwards compatibility.

Of course, related is what dimensions the data should be to support PET data for the future, in particular dynamic/gated data.

**Comment by evgueni-ovtchinnikov** [comment]
will `SegmentBySinogram<float>::full_iterator` work in the same way?

**Comment by NikEfth** [comment]
Hi,
I fixed a bug that I introduced on my last commit. And few other minor. I
will push tonight. There is an open issue with some hdf5 stuff that leaked
in. @Kris, advice on this?
Cheers

**Comment by KrisThielemans** [comment]
> will `SegmentBySinogram<float>::full_iterator` work in the same way?

yes, but currently we need another loop over tof-bins (not just views/segments), i.e. our `Segment`/Sinogram/Viewgram` classes are the same as before (just also track the TOF bin). Very easy, but obviously it cannot be done on SIRF master yet. would need a separate branch for this.

@ALEXJAZZ008008 will help with this.

**Comment by KrisThielemans** [comment]
@NikEfth let's keep STIR discussions to the STIR PR. But we cannot really remove hdf5-related commits/files, unless we go with `git filterbranch`, and screw up everybody's repository (which we don't want to do!). It is probably a matter of first merging the non-TOF GE Signa stuff.

**Comment by KrisThielemans** [comment]
See #359 for preparation

**Comment by KrisThielemans** [comment]
Probable list of things that need to change

- [ ] `cstir.cpp` `cSTIR_getAcquisitionsDimensions` needs to get number of TOF bins from STIR
- [ ] `STIR.py` `PETAcquisitionData.show()` has sinogram selection. Would need TOF bin selection.
- [ ] similar for MATLAB
- [ ] `stir_x.cpp`:  `estimate_randoms`  will need an extra loop over tof_bins (simply accumulating in the same non-TOF array)
- [ ] `stir_data_contains.cpp`
   -	PETAcquisitionData::norm() etc needs extra loop over tof bins

**Comment by KrisThielemans** [comment]
randoms still to do
